### DISCUSSION TITLE
result.orig_shape does not match the input shape

### DISCUSSION THREAD

**Discussion by Padmaapparao** [ultralytics/ultralytics#7746] [General]
prediction prints:
image 1/1 24.jpeg: 640x480 1 person, 1 suitcase, 22.8ms
Speed: 1.8ms preprocess, 22.8ms inference, 0.7ms postprocess per image at shape (1, 3, 640, 480),
When we extract orig_shape, it shows (262, 192)
How can I get the original input image shape?

**Comment by pderrenger** [comment]
@Padmaapparao the `orig_shape` attribute should reflect the original shape of the images before they were resized for inference. If you're seeing a discrepancy, it might be due to the way the images are being loaded or processed before prediction.

To ensure you're getting the correct original shape, you can manually store the shape of the images as you load them for prediction. Here's a concise example of how you might do this:

```python
from ultralytics import YOLO
import cv2

# Load your model
model = YOLO('yolov8n.pt')

# Load your image
image_path = 'path/to/your/image.jpg'
image = cv2.imread(image_path)

# Store original shape
orig_shape = image.shape[:2]  # (height, width)

# Perform prediction
results = model.predict(source=image)

# Now orig_shape contains the original image shape
print(f'Original image shape: {orig_shape}')
```

In this example, `orig_shape` will contain the height and width of the original image before any resizing done by the model's preprocessing steps. If you need to access this information after the prediction, make sure to store it as shown above.
